help/en/html/doc/Technical/TechRoadMap.shtml

```diff
@@ -155,11 +163,23 @@
       <p>In Spring 2022 the Java version required by JMRI moved forward, making
       <a href="https://www.jmri.org/releasenotes/jmri4.27.1.shtml">JMRI 4.27.1</a> the last JMRI version
       running on Java 8.<br>
-      Newer development releases 4.99.x will culminate in the June 2022 production releases
+      Newer development releases 4.99.x culminated in the June 2022 production releases
       requiring Java 11 and therefore being called JMRI 5.0.</p>
 
-      <p>Java 17 is the current LTS release.  JMRI is built and tests are run on it too,
-      but we don't recommend using it due to some remaining compatibility issues.</p>
+      <p>Java 17 is the LTS release after Java 11.  
+      Starting with the JMRI 5.9.* development series, 
+      Java 11 is still supported but Java 17 is recommended.
+      We recommend Java 17 to move the user installations forward for some future
+      migration, and because it's easier to install Java 17 on some OSs.</p>
+      
+      <p>Java 21 is the LTS release after Java 17.  We do not yet
+      recommend it becasuse of some not-fully-understood compatibility
+      edge cases. It works for some people, but we're not yet certain
+      that it will work for all.
```

Comment:
> becasuse of some not-fully-understood compatibility edge cases

Do we know which these issues are?

Reply:
Do we have a list of these edge cases?

Reply:
> Do we have a list of these edge cases?

Not really.  IIRC, there have been some printing problems and some display artifacts, both on Windows that were fixed by rolling back from Java 21.  But that's not really conclusive as to what was really going on.